services/provider/client/client.go

```diff
@@ -20,8 +22,12 @@ func NewProviderClient(ctx context.Context, serverAddr string, timeout time.Dura
 	apiCtx, cancel := context.WithTimeout(ctx, timeout)
 	defer cancel()
 
+	config := &tls.Config{
+		InsecureSkipVerify: true,
+	}
+
```

Comment:
Is this a temp fix? Disabling security checks is dangerous and should be avoided? why not use valid certificates?

Reply:
We are not sharing certs between clients and the server. So I think we are ok to skip this verification on the client side. Agree this is not very secure.   Also [see](https://github.com/red-hat-storage/ocs-operator/issues/1515#issuecomment-1037383999).

Reply:
From the official doc 
```
// InsecureSkipVerify controls whether a client verifies the server's
// certificate chain and host name. If InsecureSkipVerify is true, crypto/tls
// accepts any certificate presented by the server and any host name in that
// certificate. In this mode, TLS is susceptible to machine-in-the-middle
// attacks unless custom verification is used. This should be used only for
// testing or in combination with VerifyConnection or VerifyPeerCertificate.
InsecureSkipVerify [bool](https://pkg.go.dev/builtin#bool)
```

Reply:
Yes.  I pasted the same comment in the description, but skipped the last couple of lines. ;)